ui/component/livestreamChatLayout/view.jsx

```diff
@@ -0,0 +1,375 @@
+      doSuperChatList();
+    }
+  }, [claimId, doCommentList, doSuperChatList]);
+
+  // Register scroll handler (TODO: Should throttle/debounce)
+  React.useEffect(() => {
+    function handleScroll() {
+      if (discussionElement) {
+        const scrollTop = discussionElement.scrollTop;
+        if (scrollTop !== scrollPos) {
+          setScrollPos(scrollTop);
+        }
+      }
+    }
+
+    if (discussionElement) {
+      discussionElement.addEventListener('scroll', handleScroll);
+      return () => discussionElement.removeEventListener('scroll', handleScroll);
+    }
+  }, [discussionElement, scrollPos, viewMode]);
+
+  // Retain scrollPos=0 when receiving new messages.
+  React.useEffect(() => {
+    if (discussionElement && commentsLength > 0) {
+      // Only update comment scroll if the user hasn't scrolled up to view old comments
+      if (scrollPos >= 0) {
+        // +ve scrollPos: not scrolled (Usually, there'll be a few pixels beyond 0).
+        // -ve scrollPos: user scrolled.
+        const timer = setTimeout(() => {
+          // Use a timer here to ensure we reset after the new comment has been rendered.
+          discussionElement.scrollTop = 0;
+        }, COMMENT_SCROLL_TIMEOUT);
+        return () => clearTimeout(timer);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [commentsLength]); // (Just respond to 'commentsLength' updates and nothing else)
+
+  // Stop spinner for resolving superchats
+  React.useEffect(() => {
+    if (resolvingSuperChats) {
+      // The real solution to the sluggishness is to fix the claim store/selectors
+      // and to paginate the long superchat list. This serves as a band-aid,
+      // showing a spinner while we batch-resolve. The duration is just a rough
+      // estimate -- the lag will handle the remaining time.
+      const timer = setTimeout(() => {
+        setResolvingSuperChats(false);
+        // Scroll to the top:
+        if (discussionElement) {
+          const divHeight = discussionElement.scrollHeight;
+          discussionElement.scrollTop = divHeight * -1;
+        }
+      }, 1000);
+      return () => clearTimeout(timer);
+    }
+  }, [discussionElement, resolvingSuperChats]);
+
+  if (!claim) return null;
+
+  const chatContentToggle = (toggleMode: string, label: any) => (
+    <Button
+      className={classnames('button-toggle', { 'button-toggle--active': viewMode === toggleMode })}
+      label={label}
+      onClick={() => {
+        if (toggleMode === VIEW_MODES.SUPERCHAT) {
+          toggleSuperChat();
+        } else {
+          setViewMode(VIEW_MODES.CHAT);
+        }
+
+        if (discussionElement) {
+          const divHeight = discussionElement.scrollHeight;
+          discussionElement.scrollTop = toggleMode === VIEW_MODES.CHAT ? divHeight : divHeight * -1;
+        }
+      }}
+    />
+  );
+
+  if (openedPopoutWindow || chatHidden) {
+    return (
+      <div className="card livestream__chat">
```

Comment:
Enhancement: Some gap between these 2 cards would be nice

![image](https://user-images.githubusercontent.com/64950861/149287597-c924bb41-d134-44b4-8aa0-29ba34196269.png)